qutip/cy/cqobjevo.pyx

```diff
@@ -224,20 +232,32 @@ cdef class CQobjEvo:
 
     cpdef complex expect(self, double t, complex[::1] vec):
         if self.super:
+            if self.shape1 != vec.shape[0]:
+                raise ValueError(f"Shape don't match: ({self.shape0}, "
+                                 f"{self.shape1}) @ ({vec.shape[0]}, 1) ")
```

Comment:
The `@` sign in the error messages here suggests that the error occurred during multiplication, but it actually occurred during a call to expect. It would be good if the error message reflected where the error occurred.

We should also remove the space at the end of the string.